client/state/data-layer/utils.js

```diff
@@ -31,7 +31,8 @@ export function convertKeysBy( obj, fn ) {
 			obj,
 			( result, value, key ) => {
 				const newKey = fn( key );
-				const newValue = isObjectLike( value ) ? convertKeysBy( value, fn ) : value;
+				const newValue =
+					value !== null && typeof value === 'object' ? convertKeysBy( value, fn ) : value;
```

Comment:
Could we remove the check and call simply:
```js
const newValue = convertKeysBy( value, fn );
```
? The `convertKeysBy` function does an `_.isPlainObject` check internally and returns `value` unchanged if it's not an array or a plain object.

Reply:
That makes sense. I wonder why the `isObjectLike()` was there in the first place 🤔

Reply:
Addressed in https://github.com/Automattic/wp-calypso/pull/52419/commits/29cd6e236e1a37a4441284aecea321ba103c2c8e